src/log-store/src/kafka/util/record.rs

```diff
@@ -0,0 +1,608 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use rskafka::record::Record as KafkaRecord;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, OptionExt, ResultExt};
+
+use crate::error::{
+    ChecksumMismatchedSnafu, DecodeJsonSnafu, EmptyEntriesSnafu, EncodeJsonSnafu, GetClientSnafu,
+    IllegalSequenceSnafu, MissingKeySnafu, MissingValueSnafu, ProduceRecordSnafu, Result,
+};
+use crate::kafka::client_manager::ClientManagerRef;
+use crate::kafka::util::offset::Offset;
+use crate::kafka::{EntryId, EntryImpl, NamespaceImpl};
+
+/// The current version of Record.
+pub(crate) const VERSION: u32 = 0;
+
+/// The estimated size in bytes of a serialized RecordMeta.
+/// A record is guaranteed to have sizeof(meta) + sizeof(data) <= max_batch_size - ESTIMATED_META_SIZE.
+const ESTIMATED_META_SIZE: usize = 256;
+
+/// The type of a record.
+///
+/// If the entry is able to fit into a Kafka record, it's converted into a Full record.
+/// If the entry is too large to fit into a Kafka record, it's converted into a collection of record.
+/// Those records must contain exactly one First record and one Last record, and potentially several
+/// Middle record. There may be no Middle record.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum RecordType {
+    /// The record is self-contained, i.e. an entry's data is fully stored into this record.
+    Full,
+    /// The record contains the first part of an entry's data.
+    First,
+    /// The record contains one of the middle parts of an entry's data.
+    /// The sequence of the record is identified by the inner field.
+    Middle(usize),
+    /// The record contains the last part of an entry's data.
+    Last,
+}
+
+impl From<RecordType> for i32 {
+    fn from(tp: RecordType) -> Self {
+        match tp {
+            RecordType::Full => -1,
+            RecordType::First => 0,
+            RecordType::Middle(seq) => seq as i32,
+            RecordType::Last => i32::MAX,
+        }
+    }
+}
+
+/// The metadata of a record.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct RecordMeta {
+    /// The version of the record. Used for backward compatibility.
+    version: u32,
+    /// The type of the record.
+    pub tp: RecordType,
+    /// The id of the entry the record associated with.
+    pub entry_id: EntryId,
+    /// The namespace of the entry the record associated with.
+    pub ns: NamespaceImpl,
+    /// The computed checksum of the payload.
+    checksum: u32,
+}
+
+/// The minimal storage unit in the Kafka log store.
+/// Our own Record will be converted into a Kafka record during producing.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
```

Comment:
Maybe remove Serialize, Deserialize
```suggestion
#[derive(Debug, Clone, PartialEq)]
```